src/kbd.rs

```diff
@@ -0,0 +1,274 @@
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x60
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x70
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x80
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x90
+  NO,   NO,   NO,   NO,   NO,   CTL,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xA0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xB0
+  ALT,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xC0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xD0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xE0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xF0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+];
+
+static TOGGLECODE: [u8; 256] = [
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x00
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x10
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x20
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x30
+  NO,   NO,   CAPSLOCK,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NUMLOCK,   SCROLLLOCK,   NO,   // 0x40
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x50
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x60
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x70
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x80
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0x90
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xA0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xB0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xC0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xD0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xE0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,   // 0xF0
+  NO,   NO,   NO,   NO,   NO,   NO,   NO,   NO,
+];
+
+extern "C" {
+    fn consoleintr(c: unsafe extern fn() -> i32);
```

Comment:
This function should be available in `console.rs`.

`use console::consoleintr;`